Sources/Starknet/Accounts/StarknetAccount.swift

```diff
@@ -16,7 +16,7 @@ public class StarknetAccount: StarknetAccountProtocol {
     public let chainId: StarknetChainId
 
     private let signer: StarknetSignerProtocol
-    private let provider: StarknetProviderProtocol
+    public let provider: StarknetProviderProtocol
```

Comment:
note: `provider` needs to be changed to public, because it's added in `StarknetAccountProtocol` [here](https://github.com/software-mansion/starknet.swift/pull/217/files#diff-60a23c6840c8e03b65142bc074f2e9dd421934da6992b2f30a94cadbbfa4a6a0R8-R9).

Reply:
Hmmm, I understand that it's used in the extensions, but I wonder if `provider` should be included in the account protocol, as it's making the protocol reliant on component that is more implementation-specific (rpc communication), therefore making the account protocol less abstract

Reply:
> as it's making the protocol reliant on component that is more implementation-specific (rpc communication), therefore making the account protocol less abstract

I agree 👍 Just one remark - not adding `provider` in `StarknetAccountProtocol` enforces to add extra `provider` param in `estimateFeeV1` and `estimateFeeV3`. Hope this one is not too cumbersome while using.

Wdyt guys @DelevoXDG @mluisbrown?

Reply:
We would only need to pass in the `provider` in the `estimate` methods that don't take `nonce` param, right? That seems ok to me.

Reply:
We would keep the `provider` property in `StarknetAccount` though, right? It doesn't need to be in the `protocol` but it needs to be a public property in `StarknetAccount`.

Reply:
Can't we just make the methods abstract  and define them in `StarknetAccount` instead?